miniapps/nurbs/nurbs_naca_cmesh.cpp

```diff
@@ -653,7 +646,7 @@ double NACA4::xl(double l) const
    return x;
 }
 
-void GetTipXY(const NACA4 &foil_section, KnotVector *kv, double tf,
+void GetTipXY(const NACA4 &foil_section, unique_ptr <KnotVector> &kv, double tf,
```

Comment:
For clarity, one can make the parameter `const unique_ptr` -- this way one can still call non-const methods of `KnotVector` but one cannot extract/move out the pointer from the object referenced by `kv`. Without the `const` the method can "steal" the pointer from `kv`.

Reply:
Actually, if there is no transfer of ownership of the `KnotVector` pointed to by `kv`, it is much cleaner to just use parameter `KnotVector &` (or `const KnotVector &`, if the object is not being modified) instead of (`const`) `unique_ptr<KnotVector> &`.

Reply:
And finally, I see that I've been commenting on outdated code and the latest version is actually using `const KnotVector &`. 🤣